packages/core/test/app/db/range-bucket.spec.ts

```diff
@@ -256,4 +269,32 @@ describe('RangeDB', () => {
       ]
     )
   })
+
+  describe('iterator()', () => {
+    it('allows nextRange() to be called by the iterator returning a RangeEntry instead of a KV', async () => {
+      const testRanges = {
```

Comment:
I might suggest having the inputs be the expected results. If `db.put(...)` doesn't properly  overwrite range 50-150, this test will fail, even though the `iterator()` call works properly.